src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/core/SplashScreenResource.kt

```diff
@@ -62,20 +62,83 @@ class SplashScreenResource(private val service: SplashScreenService) {
+    description = "Retrieves a list of blocked prison ids for the screen (module) name or **ALL* if all prisons, or an empty list if there are none blocked. Requires role NOMIS_SCREEN_ACCESS",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "OK",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Bad request",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires role NOMIS_SCREEN_ACCESS",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Not found if the screen (module) name does not exist",
+        content = [
+          Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class)),
+        ],
+      ),
+    ],
+  )
+  fun getBlockedPrisons(
+    @Schema(description = "The name of the screen (module)", example = "OIDINCRS")
+    @PathVariable moduleName: String,
+  ): List<PrisonDto> = service.getBlockedPrisons(moduleName)
 }
 
+data class PrisonDto(
+  @Schema(description = "The prisonId or **ALL**", example = "MDI but can be **ALL** for all")
+  val prisonId: String,
+)
+
+@Schema(description = "A Splash condition details")
+data class SplashConditionDto(
+  @Schema(description = "The prisonId or **ALL**", example = "MDI but can be **ALL** for all")
+  val prisonId: String,
+  @Schema(description = "Whether access to the screen is blocked", example = "true")
+  val accessBlocked: Boolean,
+  @Schema(description = "The type of condition set on the screen", example = "CASELOAD")
+  val type: CodeDescription,
+)
+
 @Schema(description = "A Splash screen details")
 data class SplashScreenDto(
   @Schema(description = "The name of the module", example = "OIDINCRS")
   val moduleName: String,
 
-  @Schema(description = "The type of access")
-  val blockAccessCode: CodeDescription,
+  @Schema(description = "The type of access - YES, NO, COND")
+  val accessBlockedType: CodeDescription,
 
   @Schema(description = "The text shown when a screen is accessible but will shortly be turned off", example = "This screen will be turned off next month.")
   val warningText: String?,
 
   @Schema(description = "The text shown when a screen is blocked", example = "This screen is no longer accessible, use DPS.")
   val blockedText: String?,
+
+  @Schema(description = "The prison code", example = "BXI")
```

Comment:
Minor but I don't think this matches the field type?

Reply:
ah yes - cut and paste - will also fix in next PR